lib/pbench/test/unit/agent/test_tool_group.py

```diff
@@ -255,3 +267,56 @@ def mock_read_text(path):
             assert tg.get_label(host) == labels[host]
         assert tg.get_tools("bad-host") == {}
         assert tg.get_label("bad-host") == ""
+
+
+def test_gen_tool_groups(monkeypatch):
+    """Verify the tool group directory generator."""
+
+    class MockPath:
+        """A simple ToolGroup mock to generate host names from different
+        tool group directories.
+        """
+
+        # The path name determines the behavior.
+        NO_TOOL_GROUP_DIRS = "ntgd"
+        # Tool group names that will be found by the `.glob()` method.
+        _names = {"heavy", "light", "medium"}
```

Comment:
It's a minor point but I believe it is still the case that the config file uses `light`, `medium` and `heavy` as tool-set names, so using them as tool group names here might be a bit confusing.  It is likely of course that the tool set names (as well as the config file) will fall off by the wayside, so who cares? But I just wanted to point it out. It's definitely not something to delay the merge of this PR, so I'll resolve the convo preemptively.